front/src/modules/rollingStock/hooks/useFilterRollingStock.tsx

```diff
@@ -0,0 +1,202 @@
+import { useState, useEffect } from 'react';
+
+import { enhancedEditoastApi } from 'common/api/enhancedEditoastApi';
+import type { LightRollingStock, LightRollingStockWithLiveries } from 'common/api/osrdEditoastApi';
+import { setFailure } from 'reducers/main';
+import { useAppDispatch } from 'store';
+import { castErrorToFailure } from 'utils/error';
+// text: a string to search in the rolling stock name, detail, reference, series, type, grouping
+// elec: true if the rolling stock has an electric mode
+// thermal: true if the rolling stock has a thermal mode
+// locked: true if the rolling stock is native in the database, can't be updated/deleted
+// notLocked: true if the rolling stock is created by the user, can be updated/deleted
+export interface RollingStockFilters {
+  text: string;
+  elec: boolean;
+  thermal: boolean;
+  locked: boolean;
+  notLocked: boolean;
+}
+
+export function rollingStockPassesEnergeticModeFilters(
+  modes: LightRollingStock['effort_curves']['modes'],
+  { elec, thermal }: RollingStockFilters
+) {
+  if (elec || thermal) {
+    const effortCurveModes = Object.values(modes).map(({ is_electric: isElec }) => isElec);
+    const hasAnElectricMode = effortCurveModes.includes(true);
+    const hasAThermalMode = effortCurveModes.includes(false);
+    if ((elec && !hasAnElectricMode) || (thermal && !hasAThermalMode)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function rollingStockPassesSearchedStringFilter(
+  name: string,
+  metadata: LightRollingStock['metadata'],
+  filters: RollingStockFilters
+) {
+  if (!filters.text) {
+    return true;
+  }
+  function includesSearchedString(str: string | undefined) {
+    return str && str.toLowerCase().includes(filters.text);
+  }
+  return [
+    name,
+    metadata?.detail,
+    metadata?.reference,
+    metadata?.series,
+    metadata?.type,
+    metadata?.grouping,
+  ].some(includesSearchedString);
+}
+
+function rollingStockPassesLockedFilter(locked: boolean, filters: RollingStockFilters) {
+  if (filters.locked && !locked) {
+    return false;
+  }
+  return true;
+}
+
+function rollingStockPassesNotlockedFilter(locked: boolean, filters: RollingStockFilters) {
+  if (filters.notLocked && locked) {
+    return false;
+  }
+  return true;
+}
+
+function filterRollingStocks(
+  rollingStockList: LightRollingStockWithLiveries[],
+  filters: RollingStockFilters
+) {
+  return rollingStockList.filter(({ name, metadata, effort_curves: effortCurves, locked }) => {
```

Comment:
(only suggestion) I would rather keep rollingStock, it's clearer
```suggestion
  return rollingStockList.filter((rollingStock) => {
```